### PR TITLE
Add cluster command to show cluster and peer statuses.

### LINKED ISSUE #2192: Add command to amtool to view the cluster status
**What did you do?**
I install alertmanager cluster with three nodes on kubernetes

**What did you expect to see?**
I want to see node status in cluster by some client command, in order to jutify cluster is normal as pod health check

**What did you see instead? Under which circumstances?**
But with new version, I only can see cluster status on  alertmanager url, which can't be used in kubernetes

### PR DESCRIPTION
**Pull request by ffahri**
Closes #2192 
This pr implements amtool cluster show command

Simple output
```
$ amtool --alertmanager.url http://localhost:9094 cluster show
Cluster Status:  ready
```
Extended output
```
$ amtool --alertmanager.url http://localhost:9094 cluster show -o extended
Cluster Status:  ready

Address            Name
192.30.10.111:9094  01E4R8TX0WTNW1JYH3MB0NGHA1
192.30.42.5:9094    01E3S5C0DTS3RMJZ71RH8LPD69
192.30.42.140:9094  01E3S5BA4TX4KR030ZXXKEA6WC
```
Json output
```
$amtool --alertmanager.url http://localhost:9094 cluster show -o json
{"name":"01E3S5BY4GX4KR034ZXXREA6WC","peers":[{"address":"192.30.42.140:9094","name":"01E3S5BA4TX4KR030ZXXKEA6WC"},{"address":"192.30.42.5:9094","name":"001E3S5C0DTS3RMJZ71RH8LPD69"},{"address":"192.30.10.111:9094","name":"01E4R8TX0WTNW1JYH3MB0NGHA1"}],"status":"ready"}
```

### CODE REVIEW

File: cli/cluster.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 Prometheus Team
```
Comment (simonpasquier) [inline cli/cluster.go:1]: 2020 :)

File: cli/cluster.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	"github.com/prometheus/alertmanager/cli/format"
+)
+
+const clusterHelp = `View cluster status.
+
+The amount of output is controlled by the output selection flag:
+	- Simple: Print just cluster status
+	- Extended: Print the peers and their names with cluster status
+	- Json: Print entire cluster status object as json
```
Comment (simonpasquier) [inline cli/cluster.go:30]: I would remove this part since the `--output` flag is supported by all commands.

File: cli/format/format_extended.go [hunk 3]
```diff
@@ -89,6 +89,23 @@ func (formatter *ExtendedFormatter) FormatConfig(status *models.AlertmanagerStat
 	return nil
 }
 
+// FormatClusterStatus formats the cluster status with peers into a readable string
```
Comment (simonpasquier) [inline cli/format/format_extended.go:92]: (nit) missing dot at the end.

File: cli/format/format_extended.go [hunk 4]
```diff
@@ -89,6 +89,23 @@ func (formatter *ExtendedFormatter) FormatConfig(status *models.AlertmanagerStat
...
+func (formatter *ExtendedFormatter) FormatClusterStatus(status *models.ClusterStatus) error {
+	w := tabwriter.NewWriter(formatter.writer, 0, 0, 2, ' ', 0)
+	fmt.Fprintf(w, "Cluster Status:\t%s\n\n", *status.Status)
```
Comment (simonpasquier) [inline cli/format/format_extended.go:95]: I would add the node's name (e.g. `status.Name`).

File: cli/format/format_simple.go [hunk 5]
```diff
@@ -75,6 +75,12 @@ func (formatter *SimpleFormatter) FormatConfig(status *models.AlertmanagerStatus
 	return nil
 }
 
+func (formatter *SimpleFormatter) FormatClusterStatus(status *models.ClusterStatus) error {
+	w := tabwriter.NewWriter(formatter.writer, 0, 0, 2, ' ', 0)
+	fmt.Fprintf(w, "Cluster Status:\t%s\n", *status.Status)
```
Comment (simonpasquier) [inline cli/format/format_simple.go:80]: ditto

File: cli/format/sort.go [hunk 6]
```diff
@@ -34,3 +35,11 @@ func (s ByStartsAt) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s ByStartsAt) Less(i, j int) bool {
 	return time.Time(*s[i].StartsAt).Before(time.Time(*s[j].StartsAt))
 }
+
+type ByAddress []*models.PeerStatus
+
+func (s ByAddress) Len() int      { return len(s) }
+func (s ByAddress) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s ByAddress) Less(i, j int) bool {
+	return bytes.Compare([]byte(*s[i].Address), []byte(*s[j].Address)) < 0
```
Comment (simonpasquier) [inline cli/format/sort.go:44]: not sure why you cast string to bytes.

Comment (ffahri) [reply cli/format/sort.go]: I double-checked this logic and that was wrong. I need to collect an IP addresses from host:ip format. Then using net.IP to comparing using bytes will be much correct

Comment (ffahri) [reply cli/format/sort.go]: Here is the new way
``` 
func (s ByAddress) Less(i, j int) bool {
	ip1, _, _ := net.SplitHostPort(*s[i].Address)
	ip2, _, _ := net.SplitHostPort(*s[j].Address)
	return bytes.Compare(net.ParseIP(ip1), net.ParseIP(ip2)) < 0
}
```
Im not sure about the ignoring errors what do you think about this implementation

File: cli/format/format_extended.go (hunk 4 above)
Comment (ffahri) [reply cli/format/format_extended.go]: I added the node name into the simple format as well

File: cli/format/format_extended.go (hunk 3 above)
Comment (ffahri) [reply cli/format/format_extended.go]: added 🎉

File: cli/format/sort.go (hunk 6 above)
Comment (ffahri) [reply cli/format/sort.go]: However this still discards
127.0.0.1:1000
127.0.0.1:2000
127.0.0.1:3000

What do you think about this?
Do we sort by ip address if yes should we add ports as well ? or discards sorting ?

**Comment by ffahri** [comment]
Which part is this nit @simonpasquier  🤔

File: cli/format/sort.go [hunk 7]
```diff
@@ -34,3 +36,17 @@ func (s ByStartsAt) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
...
+
+type ByAddress []*models.PeerStatus
+
+func (s ByAddress) Len() int      { return len(s) }
+func (s ByAddress) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s ByAddress) Less(i, j int) bool {
+	ip1, port1, _ := net.SplitHostPort(*s[i].Address)
+	ip2, port2, _ := net.SplitHostPort(*s[j].Address)
+	if ip1 == ip2 {
+		return port1 < port2
```
Comment (simonpasquier) [inline cli/format/sort.go:48]: you need to convert them to integer for comparison
```suggestion
                p1, _ := strconv.Atoi(port1)
                p2, _ := strconv.Atoi(port2)
		return p1 < p2
```

**Comment by simonpasquier** [comment]
hmm GitHub discarded my previous comment...

File: cli/format/sort.go (hunk 7 above)
Comment (ffahri) [reply cli/format/sort.go]: sorry about that added.

**Comment by simonpasquier** [comment]
Thanks!

**Pull request merged**